lib/connect.js

```diff
@@ -104,37 +103,30 @@ function connect(url, socketOptions, openCallback) {
 
   var protocol, fields;
   if (typeof url === 'object') {
-    protocol = (url.protocol || 'amqp') + ':';
+    protocol = url.protocol || 'amqp:';
     sockopts.host = url.hostname;
     sockopts.servername = url.hostname;
     sockopts.port = url.port || ((protocol === 'amqp:') ? 5672 : 5671);
 
     var user, pass;
     // Only default if both are missing, to have the same behaviour as
     // the stringly URL.
-    if (url.username == undefined && url.password == undefined) {
+    if (!url.username && !url.password) {
       user = 'guest'; pass = 'guest';
     } else {
       user = url.username || '';
       pass = url.password || '';
     }
 
-    var config = {
-      locale: url.locale,
-      channelMax: url.channelMax,
-      frameMax: url.frameMax,
-      heartbeat: url.heartbeat,
-    };
-
-    fields = openFrames(url.vhost, config, sockopts.credentials || credentials.plain(user, pass), extraClientProperties);
+    fields = openFrames(url.vhost, url.searchParams, sockopts.credentials || credentials.plain(user, pass), extraClientProperties);
```

Comment:
CAUTION (maybe) `vhost` used here is not a part of a `URL` instance. 
Then again it wasn't a part of the result of `url-parse` either so this was `undefined` anyway. The only place I found where it wasn't `undefined` was in the [bad URL test](https://github.com/amqp-node/amqplib/blob/main/test/connect.js#L66).

Reply:
this sounds like a bug. can we fix it? `vhost` is important